src/components/input/ClearButton.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { FC, MutableRefObject } from 'react';
+
+import { CloseIcon } from '../icons';
+
+type Props = {
+  inputRef: MutableRefObject<HTMLInputElement | null>;
+};
+
+const triggerNativeEventFor = (
+  elm: Element,
+  { event, ...valueObj }: Record<string, unknown> & { event: Event['type'] },
+) => {
+  if (!(elm instanceof Element)) {
+    throw new Error(`Expected an Element but received ${elm} instead!`);
+  }
+
+  const [prop, value] = Object.entries(valueObj)[0] ?? [];
+  const desc = Object.getOwnPropertyDescriptor(
+    Object.getPrototypeOf(elm),
+    prop,
+  );
+
+  desc?.set?.call(elm, value);
+  elm.dispatchEvent(new Event(event, { bubbles: true }));
+};
+
+const ClearButton: FC<Props> = ({ inputRef }) => (
+  <button
+    role="button"
+    onClick={() => {
+      if (!inputRef.current) return;
+
+      triggerNativeEventFor(inputRef.current, {
+        event: 'input',
+        value: '',
+      });
+      inputRef.current.focus();
+    }}
```

Comment:
This has to trigger an `onChange` event for the controlled input to work

Reply:
I am not suer I can follow, what prevents us form passing a reset function to the component?

Reply:
We can pass this to the reset button instead of the input reference, but it doesn't change the implementation.
Because we have (externally) controlled and uncontrolled input variants, we can't rely on just modifying the internal component state, we need to make sure that the `onChange` event is fired. We can achieve this by adding an update listener to the internal state in a `useEffect` hook or we can trigger an event directly here. I went for this because it doesn't drastically change the component behaviour